frontend/src/pages-tests/__tests__/playwright/task/create-task.spec.tsx

```diff
@@ -0,0 +1,115 @@
+import { ExistingTaskDto, TaskType } from "@/api/collimator/generated/models";
+import { signInAndGotoPath } from "../authentication/authentication-helpers";
+import { expect, mockUrlResponses, test } from "../helpers";
+import { TaskListPageModel } from "./task-list-page-model";
+import {
+  getTasksControllerFindAllV0ResponseMock,
+  getTasksControllerFindOneV0ResponseMock,
+  getTasksControllerUpdateFileV0ResponseMock,
+} from "@/api/collimator/generated/endpoints/tasks/tasks.msw";
+import {
+  getTasksControllerFindAllV0Url,
+  getTasksControllerFindOneV0Url,
+} from "@/api/collimator/generated/endpoints/tasks/tasks";
+import { initialDummyTask, routeDummyApp } from "./helpers";
+import { TaskFormPageModel } from "./task-form-page-model";
+import multipart from "parse-multipart-data";
+
+const task: ExistingTaskDto = getTasksControllerFindOneV0ResponseMock();
+
+const mockTasks = [
+  ...getTasksControllerFindAllV0ResponseMock().slice(0, 9),
+  task,
+];
+
+const updatedTask: ExistingTaskDto = {
+  ...getTasksControllerUpdateFileV0ResponseMock(),
+  id: task.id,
+  type: TaskType.SCRATCH,
+};
+
+let createTaskRequest: Buffer | null = null;
+let createTaskRequestHeaders: {
+  [key: string]: string;
+} | null = null;
+
+test.describe("/task/[taskId]/edit", () => {
+  test.beforeEach(async ({ page, baseURL, apiURL, scratchURL }) => {
+    createTaskRequest = null;
+    createTaskRequestHeaders = null;
+
+    await mockUrlResponses(
+      page,
+      `${apiURL}${getTasksControllerFindAllV0Url()}`,
+      {
+        get: mockTasks,
+        post: updatedTask,
+      },
+      {
+        post: (request) => {
+          createTaskRequest = request.postDataBuffer();
+          createTaskRequestHeaders = request.headers();
+        },
+      },
+    );
+
+    await mockUrlResponses(
+      page,
+      `${apiURL}${getTasksControllerFindOneV0Url(task.id)}`,
+      {
+        get: task,
+      },
+      {},
+    );
+
+    await routeDummyApp(page, `${scratchURL}/edit`);
+
+    await signInAndGotoPath(page, baseURL!, `/task`);
+
+    const list = await TaskListPageModel.create(page);
+    await list.createItem();
+  });
+
+  test("can create a task", async ({ page: pwPage, baseURL }) => {
```

Comment:
Note to self:
- Happy path testing 👍 
- Missing non-happy paths - e.g. can't submit without a file, without a title, etc.)
- Missing exception flows - e.g. network gets interrupted half-way